main.go

```diff
@@ -269,11 +279,86 @@ func (fp *Firehose) RunLeader(ctx context.Context) {
 	}
 }
 
+func (fp *Firehose) IsLeader(ctx context.Context) (*bool, error) {
```

Comment:
why `*bool` and not just plain `bool` ?

Reply:
when error is returned is it okay if bool is false? Not sure what the convention is for that since false is sort of a valid return I guess err is assumed to always be checked?

Reply:
Yes, in case of errors, zero value of the type ought to be returned.